File: backend/src/models/base_geral.py
from datetime import date
from typing import Optional

from sqlmodel import Field, SQLModel


class BaseGeralModel(SQLModel, table=True):
    __tablename__ = "base_geral"  # type: ignore
    __table_args__ = {"schema": "central_teste"}

    id: Optional[int] = Field(default=None, primary_key=True)
    ticket: Optional[int] = None
    solicitante: Optional[str] = None
    data_solicitacao: Optional[date] = None
    etapa_atual: Optional[str] = None
    status: Optional[str] = None
    nome_colaborador: Optional[str] = None
    tipo_exame: Optional[str] = None
    data_exame: Optional[date] = None
    cidade_preferencia: Optional[str] = None
    agendar_enquadramento_pcd: Optional[str] = None
    possui_formulario_rac: Optional[str] = None
    possui_exames_complementares: Optional[str] = None
    data_conclusao_dia: Optional[date] = None
    status_sla_agendamento: Optional[str] = None
    data_ultima_alteracao: Optional[date] = None
    dados_agendamento_tipo_solicitacao: Optional[str] = None

File: frontend/src/@types/baseGeral.ts
export type BaseGeral = {
	id: number;
	ticket: number | null;
	solicitante: string | null;
	data_solicitacao: string | null;
	etapa_atual: string | null;
	status: string | null;
	nome_colaborador: string | null;
	tipo_exame: string | null;
	data_exame: string | null;
	cidade_preferencia: string | null;
	agendar_enquadramento_pcd: string | null;
	possui_formulario_rac: string | null;
	possui_exames_complementares: string | null;
	data_conclusao_dia: string | null;
	status_sla_agendamento: string | null;
	data_ultima_alteracao: string | null;
	dados_agendamento_tipo_solicitacao: string | null;
};

export type BaseGeralQuery = {
	page?: number;
	size?: number;
	sort_by?: "id" | "ticket" | "data_solicitacao";
	sort_order?: "asc" | "desc";
	ticket?: number | "";
	etapa_atual?: string;
	data_solicitacao__gte?: string;
	data_solicitacao__lte?: string;
	solicitante?: string;
	status?: string;
	nome_colaborador?: string;
	tipo_exame?: string;
};
